tests/models/siglip/test_modeling_siglip.py

```diff
@@ -191,19 +191,15 @@ def test_model(self):
         config_and_inputs = self.model_tester.prepare_config_and_inputs()
         self.model_tester.create_and_check_model(*config_and_inputs)
 
-    @unittest.skip(reason="SiglipVisionModel does not support standalone training")
     def test_training(self):
         pass
```

Comment:
This isn't quite right - this is just removing the skip reason. What we want to do is reenable the tests. This means removing this method completely, as it's overriding the mixin test

Reply:
I see. I see there are multiple instances of `test_training` overrides, and also other training tests like with GC. Should I remove all those too?

Reply:
> Should I add the loss output to the model output object on those classes?

What would the loss be for those classes? 

I think it only really makes sense to have a loss for SigLipModel and SigLipForImageClassification (which would have a cross-entropy loss). 

For the components without loss implemented, they should still skip training tests

Reply:
The ImageClassification class does have loss output (BCE/CE), but it doesn't work with gradient checkpointing as only the GC training tests are failing while the normal training test passes. Should be https://github.com/huggingface/transformers/pull/27124 since CLIP has the same issue and their arch are same. Thus I will disable GC training tests too.

I don't know why GC training works for `SigLipModel` and `SiglipVisionModel` but not `ImageClassification` though, as `ImageClassification` is just a `SiglipVisionModel` chained with an FC classifier layer.

```
=================================== FAILURES ===================================
__ SiglipForImageClassificationModelTest.test_training_gradient_checkpointing __
[gw0] linux -- Python 3.10.14 /usr/local/bin/python3
tests/test_modeling_common.py:938: in test_training_gradient_checkpointing
    self.check_training_gradient_checkpointing()
tests/test_modeling_common.py:913: in check_training_gradient_checkpointing
    self.assertTrue(v.grad is not None, f"{k} in {model_class.__name__} has no gradient!")
E   AssertionError: False is not true : vision_model.head.probe in SiglipForImageClassification has no gradient!
_ SiglipForImageClassificationModelTest.test_training_gradient_checkpointing_use_reentrant _
[gw0] linux -- Python 3.10.14 /usr/local/bin/python3
tests/test_modeling_common.py:943: in test_training_gradient_checkpointing_use_reentrant
    self.check_training_gradient_checkpointing(gradient_checkpointing_kwargs={"use_reentrant": True})
tests/test_modeling_common.py:913: in check_training_gradient_checkpointing
    self.assertTrue(v.grad is not None, f"{k} in {model_class.__name__} has no gradient!")
E   AssertionError: False is not true : vision_model.head.probe in SiglipForImageClassification has no gradient!
_ SiglipForImageClassificationModelTest.test_training_gradient_checkpointing_use_reentrant_false _
[gw0] linux -- Python 3.10.14 /usr/local/bin/python3
tests/test_modeling_common.py:948: in test_training_gradient_checkpointing_use_reentrant_false
    self.check_training_gradient_checkpointing(gradient_checkpointing_kwargs={"use_reentrant": False})
tests/test_modeling_common.py:913: in check_training_gradient_checkpointing
    self.assertTrue(v.grad is not None, f"{k} in {model_class.__name__} has no gradient!")
E   AssertionError: False is not true : vision_model.head.probe in SiglipForImageClassification has no gradient!
```

Reply:
> Thus I will disable GC training tests too but keep the normal training tests on.

OK! 

> I don't know why GC training works for SigLipModel and SiglipVisionModel but not ImageClassification though, as ImageClassification is just a SiglipVisionModel chained with an FC classifier layer.

Tbh, I have no idea. Maybe @SunMarc would know?

Reply:
Should I add a warning or exception when user tries to train with GC on an ImageClassification model? I doubt it's a popular use case because siglip is mostly used by its vision branch only for vision LLMs or for zeroshot image cls and not fixed labels image cls.

Reply:
Hi @aliencaocao, I did some research and here are my findings:
- `SiglipVisionModel` GC test is passing because it is registered as a base model and we pass the test this is the case.
- SiglipForImageClassification GC test is failing because the vision head parameter gradients are None. This is completely normal since the loss calculated at the end do not use the output of the vision head module (`last_hidden_state` is used but not `pooler_output`). Hence, the gradients are None in this module. 
- `SigLipModel` passes the GC test because the loss calculated uses both `last_hidden_state` and `pooler_output`. 

Maybe we should remove the gradient check @amyeroberts in GC tests ? I'm not sure if this is an isolated case.

Reply:
@SunMarc Thanks for the inputs, that clears things up a lot.

> we skip the test this is the case since the loss is not defined.

This still shows PASSED in CI which is weird, should probably add some logs on it being actually skipped. I will add back the `unittest.skip` for it.

RE. SiglipForImageClassification, is there a way to disable the check or override it? E.g. subclass the pipe test mixin or lower classes. If it is too much duplicated code then maybe a switch as argument will be cleaner.

Reply:
Ok, I have already done that in the previous commit

Reply:
Great - could you remove for the vision model tests too? We recently updated the test suit so we should be explicitly skipping tests now.

Reply:
I thought thats being done already： https://github.com/aliencaocao/transformers/blob/5ee9d1fe74e4469c1cd9c4bfb298f41ea740a5eb/tests/models/siglip/test_modeling_siglip.py#L194

Reply:
Ah, I see, I might have misunderstood. From the messages above, I thought that tests were being correctly controlled and skipped within the original `test_training` etc. tests and so we could remove the overrides + decorators here. In this case, we're good to merge!